apps/asap-server/src/entities/research-output.ts

```diff
@@ -67,45 +80,62 @@ export const parseGraphQLResearchOutput = (
 
+  const data = output.flatData;
+
   return {
     id: output.id,
     created: parseDate(output.created).toISOString(),
-    link: output.flatData?.link || undefined,
-    type: output.flatData?.type || 'Proposal',
-    subTypes: output.flatData?.subtype ? [output.flatData.subtype] : [],
-    title: output.flatData?.title || '',
-    description: output.flatData?.description || '',
-    tags: output.flatData?.tags || [],
-    publishDate: output.flatData?.publishDate || undefined,
-    labCatalogNumber: output.flatData?.labCatalogNumber || undefined,
-    doi: output.flatData?.doi || undefined,
-    accession: output.flatData?.accession || undefined,
-    rrid: output.flatData?.rrid || undefined,
-    addedDate: output.flatData?.addedDate || undefined,
+    link: data.link || undefined,
+    type: data.type && isResearchOutputType(data.type) ? data.type : 'Proposal',
+    subTypes:
+      data.subtype && isResearchOutputSubtype(data.subtype)
+        ? [data.subtype]
+        : [],
+    title: data.title || '',
+    description: data.description || '',
+    tags: data.tags || [],
+    publishDate: data.publishDate || undefined,
+    labCatalogNumber: data.labCatalogNumber || undefined,
+    doi: data.doi || undefined,
+    accession: data.accession || undefined,
+    rrid: data.rrid || undefined,
+    addedDate: data.addedDate || undefined,
     lastUpdatedPartial:
-      output.flatData?.lastUpdatedPartial ||
-      output.lastModified ||
-      output.created,
-    accessInstructions: output.flatData?.accessInstructions || undefined,
+      data.lastUpdatedPartial || output.lastModified || output.created,
+    accessInstructions: data.accessInstructions || undefined,
     ...optionalAuthors,
     ...optionalTeams,
-    sharingStatus: output.flatData?.sharingStatus || 'Network Only',
-    asapFunded: convertDecisionToBoolean(output.flatData?.asapFunded),
-    usedInPublication: convertDecisionToBoolean(
-      output.flatData?.usedInAPublication,
-    ),
+    sharingStatus:
+      data.sharingStatus !== null && isSharingStatus(data.sharingStatus)
+        ? data.sharingStatus
+        : 'Network Only',
+    asapFunded: convertDecisionToBoolean(data.asapFunded),
+    usedInPublication: convertDecisionToBoolean(data.usedInAPublication),
     pmsEmails: uniquePmsEmails,
   };
 };
 
 const parseGraphqlTeamLite = (
-  graphqlTeam: GraphqlTeam,
+  graphqlTeam: FetchResearchOutput_findResearchOutputsContent_referencingTeamsContents,
 ): ResearchOutputResponse['team'] => ({
   id: graphqlTeam.id,
   displayName: graphqlTeam.flatData?.displayName || '',
 });
 
 const convertDecisionToBoolean = (
-  decision?: DecisionOption | null,
+  decision: string | null,
 ): boolean | undefined =>
   decision && ['Yes', 'No'].includes(decision) ? decision === 'Yes' : undefined;
+
+const isSharingStatus = (
+  status: string,
+): status is ResearchOutputSharingStatus =>
+  (sharingStatuses as ReadonlyArray<string>).includes(status);
```

Comment:
the casting is necessary here because of how `includes` is typed. It's really odd and many people are against this sort of type definition as it kinda defeats the purpose of `includes` but oh well